src/main/java/fr/insee/pearljam/api/service/impl/SurveyUnitUpdateServiceImpl.java

```diff
@@ -22,24 +24,37 @@
 @Slf4j
 public class SurveyUnitUpdateServiceImpl implements SurveyUnitUpdateService {
 
+    private final CommunicationRequestRepository communicationRequestRepository;
+
     @Transactional
     @Override
-    public void updateSurveyUnitInfos(SurveyUnit surveyUnit, SurveyUnitDetailDto surveyUnitDetailDto) {
-        if(surveyUnitDetailDto.getComments() != null) {
-            Set<Comment> commentsToUpdate = surveyUnitDetailDto.getComments().stream()
+    public void updateSurveyUnitInfos(SurveyUnit surveyUnit, SurveyUnitUpdateDto surveyUnitUpdateDto) {
+        if(surveyUnitUpdateDto.comments() != null) {
+            Set<Comment> commentsToUpdate = surveyUnitUpdateDto.comments().stream()
                     .map(commentDto -> CommentDto.toModel(surveyUnit.getId(), commentDto))
                     .collect(Collectors.toSet());
 
             surveyUnit.updateComments(commentsToUpdate);
         }
-        if(surveyUnitDetailDto.getCommunicationRequests() != null) {
-            List<CommunicationRequest> communicationRequests = surveyUnitDetailDto.getCommunicationRequests().stream()
-                    .map(CommunicationRequestDto::toModel)
+        if(surveyUnitUpdateDto.communicationRequests() != null) {
+            List<CommunicationRequest> communicationRequestsToCreate = CommunicationRequestCreateDto.toModel(surveyUnitUpdateDto.communicationRequests());
+            Set<CommunicationRequestDB> currentCommunicationRequests = surveyUnit.getCommunicationRequests();
+            List<CommunicationRequest> newCommunicationsRequests = communicationRequestsToCreate.stream()
+                    .filter(newCommunicationRequest -> {
+                        for (CommunicationRequestDB currentCommunicationRequest : currentCommunicationRequests) {
+                            // if survey unit has already this communication request, skip it
+                            if(currentCommunicationRequest.getCommunicationTemplate().getId()
+                                    .equals(newCommunicationRequest.communicationTemplateId())) {
+                                return false;
+                            }
```

Comment:
something wrong here:
there could be multiple commRequests with the same commTemplateId
should discuss this